iree/hal/cuda/cuda_device.c

```diff
@@ -18,9 +19,10 @@
 #include "iree/hal/cuda/dynamic_symbols.h"
 #include "iree/hal/cuda/event_semaphore.h"
 #include "iree/hal/cuda/executable_layout.h"
-#include "iree/hal/cuda/graph_command_buffer.h"
```

Comment:
Leave the graph command buffer in here and make whether to use it an option. One of the points of doing all of this work is that we can compare the stream and the graph implementations and that should be easy to toggle between. You'd then have create_command_buffer return either the deferred command buffer or the graph command buffer, and during submit switch on the same option (for now assuming that it's either one or the other and not mixed types).

Reply:
gotcha, restored